src/lib.rs

```diff
@@ -2558,6 +2562,44 @@ impl Connection {
         Ok(sent)
     }
 
+    /// Sets the priority for a stream.
+    ///
+    /// A stream's priority determines the order in which stream data is sent
+    /// on the wire (streams with lower priority are sent first). Streams are
+    /// created with a default priority of `127`.
+    ///
+    /// If the provided priority matches the stream's existing priority,
+    /// [`Done`] is returned. If the stream did not exist prior to calling
+    /// this method, [`InvalidStreamState`] is returned.
+    ///
+    /// [`Done`]: enum.Error.html#variant.Done
+    /// [`InvalidStreamState`]: enum.Error.html#variant.InvalidStreamState
+    pub fn stream_priority(
+        &mut self, stream_id: u64, priority: u64,
+    ) -> Result<()> {
+        // Get existing stream.
+        let stream = self
+            .streams
+            .get_mut(stream_id)
+            .ok_or(Error::InvalidStreamState)?;
+
+        if stream.priority == priority {
+            return Err(Error::Done);
```

Comment:
Error::Done doesn't seem super helpful for the calling app. 

With a test implementation, I'm calculating the new priority based on information in H3 layer and that might yield the same answer, but I don't store the old priority so can't check that before calling `stream_priority()`. That means instead of being able to do `conn.stream_priority(id, priority)?`, I need to capture Ok and Error::Done and throw them away.